oe_theme.theme

```diff
@@ -210,153 +210,352 @@ function oe_theme_preprocess_paragraph__oe_links_block(array &$variables): void
-  $sections_about_eu = [
-    'title' => $about_eu['title'],
-    'links' => [],
-  ];
-
-  foreach ($about_eu['items'] as $item) {
-    $sections_about_eu['links'][] = [
+  $legal_navigation = [];
+  foreach ($corporate_footer['legal_navigation'] as $link) {
+    $legal_navigation[] = [
       'link' => [
-        'label' => $item['label'],
-        'path' => $item['href'],
-        'icon_position' => 'after',
-      ],
-      'icon' => [
-        'path' => $variables['ecl_icon_path'],
-        'type' => 'ui',
-        'name' => 'external',
+        'label' => $link['label'],
+        'path' => $link['href'],
       ],
     ];
   }
+  $variables['processed']['legal_navigation'] = $legal_navigation;
+
+  $ecl_branding = theme_get_setting('branding') ?? 'core';
+  $preprocess = '_oe_theme_preprocess_oe_corporate_blocks_ec_footer__' . $ecl_branding;
+  $preprocess($variables);
+}
 
+/**
+ * Process corporate_footer data to match ecl component expected format.
+ *
+ * @param array $variables
+ *   Set of available variables.
+ *
+ * @see hook_preprocess_oe_corporate_blocks_ec_footer().
+ */
+function _oe_theme_preprocess_oe_corporate_blocks_ec_footer__core(array &$variables): void {
   $variables['corporate_footer']['sections'] = [
-    $sections_about_ec,
-    $sections_social_media,
-    $sections_about_eu,
+    [
+      'title' => $variables['processed']['site_name_link'],
+      'description' => $variables['processed']['site_owner'],
+    ],
+    [
+      'content_before' => t('More information on:'),
+      'links' => $variables['processed']['class_navigation'],
+      'list_class_name' => 'ecl-footer-core__section2 ecl-footer-core__list--columns',
+      'section_class_name' => 'ecl-footer-core__section--separator',
+    ],
+    [
+      'links' => $variables['processed']['service_navigation'],
+    ],
+    [
+      'links' => $variables['processed']['legal_navigation'],
+    ],
   ];
+}
 
-  // Create the list of bottom link.
-  foreach ($corporate_footer['bottom_links'] as $item) {
-    $variables['corporate_footer']['common'][] = [
-      'link' => [
-        'label' => $item['label'],
-        'path' => $item['href'],
+/**
+ * Process corporate_footer data to match ecl component expected format.
+ *
+ * @param array $variables
+ *   Set of available variables.
+ *
+ * @see hook_preprocess_oe_corporate_blocks_ec_footer().
+ */
+function _oe_theme_preprocess_oe_corporate_blocks_ec_footer__standardised(array &$variables): void {
+  $variables['corporate_footer']['sections'] = [
+    [
+      'title' => [
+        'link' => [
+          'label' => $variables['processed']['site_name'],
+          'path' => $variables['processed']['site_path'],
+        ]
       ],
-    ];
-  }
-
-  $variables['custom_footer'] = _oe_theme_preprocess_oe_corporate_blocks_site_specific_footer($variables);
+      'description' => $variables['processed']['site_owner'],
+      'section_id' => 1,
+      'type' => 'site_name',
```

Comment:
The type variable does nothing, we can simlpy ignore it (unless there is a reason I am not aware of)